pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1373,3 +1373,267 @@ func TestRemoveCondition(t *testing.T) {
 		}
 	}
 }
+
+func TestSlowStartBatch(t *testing.T) {
+	fn := func() error {
+		return nil
+	}
+	errFn := func() error {
+		return fmt.Errorf("fake error")
+	}
+
+	tests := []struct {
+		name                        string
+		diff                        int
+		fn                          func() error
+		expectedSuccessfulCreations int
+		expectedErr                 error
+	}{
+		{
+			"diff = 0, non-error func",
+			0,
+			fn,
+			0,
+			nil,
+		},
+		{
+			"diff = 0, error func",
+			0,
+			errFn,
+			0,
+			nil,
+		},
+		{
+			"diff = 10, non-error func",
+			10,
+			fn,
+			10,
+			nil,
+		},
+		{
+			"diff = 10, error func",
+			10,
+			errFn,
+			0,
+			fmt.Errorf("fake error"),
+		},
+	}
+
+	for _, test := range tests {
+		successfulCreations, err := slowStartBatch(test.diff, controller.SlowStartInitialBatchSize, test.fn)
+		if successfulCreations != test.expectedSuccessfulCreations {
+			t.Errorf("%s: unexpected processed batch size, expected %d, got %d", test.name, test.expectedSuccessfulCreations, successfulCreations)
+		}
+		if !reflect.DeepEqual(err, test.expectedErr) {
+			t.Errorf("%s: unexpected processed batch size, expected %v, got %v", test.name, test.expectedErr, err)
+		}
+	}
+}
+
+func TestGetPodsToDelete(t *testing.T) {
+	labelMap := map[string]string{"name": "foo"}
+	rs := newReplicaSet(1, labelMap)
+	// an unscheduled, pending pod
+	unscheduledPendingPod := newPod("unscheduled-pending-pod", rs, v1.PodPending, nil, true)
+	// an unscheduled, running, not-ready pod
```

Comment:
I don't think an unscheduled Pod could ever be Running in reality. You should only need one case for unscheduled (Pending).